### PR TITLE
Add MO extras requires for supported DL frameworks

### PR DESCRIPTION
**Pull request by slyubimt**
### Details:
 - *Add functionality to install DL framework requirements using extras tags:* `pip install openvino-mo[mxnet,tensorflow2]`
 - List of supported tags:
    - caffe
    - kaldi
    - mxnet
    - onnx
    - tensorflow
    - tensorflow2

### CODE REVIEW

File: model-optimizer/setup.py [hunk 1]
```diff
@@ -15,9 +15,13 @@
 from setuptools import setup, find_packages
 from setuptools.command.install import install
 from setuptools.command.build_py import build_py
+from pathlib import Path
```
Comment (lazarevevgeny) [inline model-optimizer/setup.py:18]: sort imports according to PEP8

File: model-optimizer/setup.py [hunk 2]
```diff
@@ -15,9 +15,13 @@
...
 
 PACKAGE_NAME = 'mo'
 
+SETUP_DIR = Path(__file__).resolve().parent
```
Comment (lazarevevgeny) [inline model-optimizer/setup.py:22]: Re-format the whole file according to PEP8

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by dkurt — APPROVED** [review]
Good feature!

File: model-optimizer/setup.py [hunk 3]
```diff
@@ -74,33 +82,42 @@ def find_package_modules(self, package, package_dir):
...
+      'mo.extensions.front.onnx': ['*.json'],
+      'mo.extensions.front.tf': ['*.json'],
+    },
+    extras_require={
+      'caffe': read_text('requirements_caffe.txt'),
+      'kaldi': read_text('requirements_kaldi.txt'),
+      'mxnet': read_text('requirements_mxnet.txt'),
+      'onnx': read_text('requirements_onnx.txt'),
+      'tensorflow': read_text('requirements_tf.txt'),
+      'tensorflow2': read_text('requirements_tf2.txt'),
```
Comment (dkurt) [inline model-optimizer/setup.py:115]: Maybe just `tf` and `tf2`?

Comment (dkurt) [reply model-optimizer/setup.py:115]: Or in addition them,
```python
      'tensorflow': read_text('requirements_tf.txt'),
      'tensorflow2': read_text('requirements_tf2.txt'),
      'tf': read_text('requirements_tf.txt'),
      'tf2': read_text('requirements_tf2.txt'),
```

Comment (slyubimt) [reply model-optimizer/setup.py:115]: these tags name will be reused in openvino-dev packages where should have consistent naming along with other components. OMZ tools already provides names for extras: 
https://github.com/openvinotoolkit/open_model_zoo/blob/71376c6ec0e485eb8c7c117e8754120d31369f67/tools/downloader/setup.py#L99

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**